### PR TITLE
DBZ-2150 Awaiting coordinator shutdown in embedded engine

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/browse/DBZ-2150

@Naros, @jpechane, could either of you review and apply? Thx!

### CODE REVIEW

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java [hunk 1]
```diff
@@ -735,6 +737,7 @@ public OffsetStorageReader offsetStorageReader() {
 
                             // Purposely not marking this method with @Override as it was introduced in Kafka 2.x
                             // and otherwise would break builds based on Kafka 1.x
+                            @Override
```
Comment (jpechane) [inline debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java:740]: Please read the comment above the newly added line

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java [hunk 2]
```diff
@@ -971,6 +974,7 @@ public boolean stop() {
         Thread thread = this.runningThread.getAndSet(null);
         if (thread != null) {
             try {
+                // Making sure the event source coordinator has enough time to shut down before forcefully stopping it
```
Comment (jpechane) [inline debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java:977]: The defautl wait time is pretty large. In this case there should be a log message informing the user about the wait time.

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java]: Good catch, my IDE automatically added it.

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java (hunk 2 above)
Comment (gunnarmorling) [reply debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java:981]: Done.

**Comment by gunnarmorling** [comment]
@jpechane, updated as per your comments.

**Comment by jpechane** [comment]
@gunnarmorling Reformatting is needed. I am with the change now, just the log message should say that the value is ms. When build passes the PR can be merged.

**Comment by gunnarmorling** [comment]
> just the log message should say that the value is ms.

It's a `Duration`, it will render with units: " PT5M".

Force-pushed for formatting.

**Comment by jpechane** [comment]
@gunnarmorling I must admit I liked the dynamically computed default timeout more. Why did you change it?

**Comment by gunnarmorling** [comment]
> Why did you change it?

Because it made for a very awkward timeout of 190 sec :)

>

**Pull request merged**